Tests/xSharePoint/xSharePoint.xSPVisioServiceApp.Tests.ps1

```diff
@@ -0,0 +1,82 @@
+[CmdletBinding()]
+param(
+    [string] $SharePointCmdletModule = (Join-Path $PSScriptRoot "..\Stubs\SharePoint\15.0.4693.1000\Microsoft.SharePoint.PowerShell.psm1" -Resolve)
+)
+
+$ErrorActionPreference = 'stop'
+Set-StrictMode -Version latest
+
+$RepoRoot = (Resolve-Path $PSScriptRoot\..\..).Path
+$Global:CurrentSharePointStubModule = $SharePointCmdletModule 
+
+$ModuleName = "MSFT_xSPVisioServiceApp"
+Import-Module (Join-Path $RepoRoot "Modules\xSharePoint\DSCResources\$ModuleName\$ModuleName.psm1")
+
+Describe "xSPVisioServiceApp" {
+    InModuleScope $ModuleName {
+        $testParams = @{
+            Name = "Test Visio App"
+            ApplicationPool = "Test App Pool"
+        }
+        Import-Module (Join-Path ((Resolve-Path $PSScriptRoot\..\..).Path) "Modules\xSharePoint")
+
+        
+        Mock Invoke-xSharePointCommand { 
+            return Invoke-Command -ScriptBlock $ScriptBlock -ArgumentList $Arguments -NoNewScope
+        }
+        
+        Import-Module $Global:CurrentSharePointStubModule -WarningAction SilentlyContinue
+
+        Context "When no service applications exist in the current farm" {
+
+            Mock Get-SPServiceApplication { return $null }
+            Mock New-SPVisioServiceApplication { }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns false when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "creates a new service application in the set method" {
+                Set-TargetResource @testParams
+                Assert-MockCalled New-SPVisioServiceApplication 
+            }
+        }
+
+        Context "When service applications exist in the current farm but the specific Visio Graphics app does not" {
+
+            Mock Get-SPServiceApplication { return @(@{
+                TypeName = "Some other service app type"
+            }) }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+        }
+
+        Context "When a service application exists and is configured correctly" {
```

Comment:
Can we put one more context in here for when the service app exists but has the wrong app pool? Then we can test that we get a false out of test-targetresource and that set-targetresource calls update appropriately